Stop spewing stacks on error

mook-as opened the pull request:
[bsc#1076242](https://bugzilla.suse.com/show_bug.cgi?id=1076242): we were dumping the stack when things go wrong, because we were using the CF CLI terminal module (which upstream had noted at some point it was not designed for), and it was no longer quieting panics as expected.

This also includes changes to make this whole package build.

mook-as committed:
Fix running lint and vet

mook-as committed:
Fix vet errors

mook-as committed:
Go import path: github.com/hpcloud/cf-plugin-usb -> SUSE/cf-usb-plugin

Otherwise things are not going to build

mook-as committed:
Make the existing tests pass

Somebody changed code but never ran the tests?

mook-as committed:
Fix crash when `cf usb` is run with no arguments

mook-as committed:
Replace CF CLI terminal library with termui

The upstream library isn't meant to be used outside of the CLI; it
panicks to handle aborts and we end up dumping the stack trace to the
user which isn't nice at all.

mook-as committed:
`usb info`: Fix crash when /info endpoint exists but wrong

If we point the USB plugin at an endpoint that _does_ respond to /info
but doesn't give us the expected versions, we can up crashing here.
This is the case if we point it at the CF API endpoint, for example.

mook-as committed:
Main: Don't spew usage help text for internal CF CLI commands

We were printing out the usage message on uninstalling the plugin.

andreas-kupries commented:
Fresh clone, first clone and checkout, then `make`:

Lint fail:

```

==>  Linting
Failed: /fast/go/src/github.com/SUSE/cf-usb-plugin/main.go:57:3: var usbUrl should be usbURL
Failed: /fast/go/src/github.com/SUSE/cf-usb-plugin/cmd/create-driver-endpoint.go:21:4: var targetUrl should be targetURL
Failed: /fast/go/src/github.com/SUSE/cf-usb-plugin/cmd/root.go:13:5: var configJson should be configJSON
Failed: /fast/go/src/github.com/SUSE/cf-usb-plugin/cmd/update-driver-endpoint.go:22:4: var targetUrl should be targetURL
Failed: /fast/go/src/github.com/SUSE/cf-usb-plugin/commands/instance.go:80:49: method parameter targetUrl should be targetURL
Failed: /fast/go/src/github.com/SUSE/cf-usb-plugin/commands/writer_unix.go:7:5: exported var Writer should have comment or be unexported
```

Test fail

```
go test -cover github.com/SUSE/cf-usb-plugin github.com/SUSE/cf-usb-plugin/cmd github.com/SUSE/cf-usb-plugin/commands github.com/SUSE/cf-usb-plugin/config github.com/SUSE/cf-usb-plugin/config/fakes github.com/SUSE/cf-usb-plugin/lib github.com/SUSE/cf-usb-plugin/lib/client/operations github.com/SUSE/cf-usb-plugin/lib/fakes github.com/SUSE/cf-usb-plugin/lib/models github.com/SUSE/cf-usb-plugin/lib/plugin
/fast/go/src/github.com/SUSE/cf-usb-plugin/vendor/github.com/SUSE/termui/error_printer.go:5:2: cannot find package "github.com/suse/termui/sigint" in any of:
	/fast/go/src/github.com/SUSE/cf-usb-plugin/vendor/github.com/suse/termui/sigint (vendor tree)
	/usr/local/go/src/github.com/suse/termui/sigint (from $GOROOT)
	/fast/go/src/github.com/suse/termui/sigint (from $GOPATH)
/fast/go/src/github.com/SUSE/cf-usb-plugin/vendor/github.com/SUSE/termui/ui.go:11:2: cannot find package "github.com/suse/termui/termpassword" in any of:
	/fast/go/src/github.com/SUSE/cf-usb-plugin/vendor/github.com/suse/termui/termpassword (vendor tree)
	/usr/local/go/src/github.com/suse/termui/termpassword (from $GOROOT)
	/fast/go/src/github.com/suse/termui/termpassword (from $GOPATH)
```

The test fail seems to be a `suse` vs `SUSE` mismatch in the go code.

mook-as commented:
Linting issues are expected; I didn't want to fix them this round because the PR is already big enough and they don't break building.
The test error isn't though; do you happen to have a `github.com/suse` (lower case) in your `GOPATH`? I'll look on my side as well.

mook-as commented:
Wait, that issue is in `termui` 😢 Let me make a PR against that…

andreas-kupries commented:
Ok re linting.

Re the suse/SUSE I see

```
./vendor/github.com/SUSE/termui/termpassword/password_reader_unix.go:14:	"github.com/suse/termui/sigint"
./vendor/github.com/SUSE/termui/error_printer.go:5:	"github.com/suse/termui/sigint"
./vendor/github.com/SUSE/termui/ui.go:11:	"github.com/suse/termui/termpassword"
```

I.e. the vendored `termui` has bad include paths, expecting `suse` in two places.
However my GOPATH has only `SUSE`

```
andreask@ten:~/fast/dev/cf-usb-plugin> find /fast/go vendor -type d -print|grep -i '/suse$'
/fast/go/src/github.com/SUSE
vendor/github.com/SUSE
```

andreas-kupries commented:
x-ed, confirming each other

viovanov commented on Godeps/Godeps.json:
```diff
@@ -13,6 +16,18 @@
 		},
+		{
+			"ImportPath": "github.com/SUSE/termui",
+			"Rev": "92556107bd806bea5d53728326e7d45c7b03f43c"
```
This is still an old commit, how come?

andreas-kupries replied:
Mark did not update this yet. Likely waited for https://github.com/SUSE/termui/pull/8 to be merged.
AFAIK currently working on updating this.

mook-as committed:
Update vendored termui

This makes sure we use upper case SUSE for go import path everywhere
https://github.com/SUSE/cf-usb-plugin/pull/1#issuecomment-359121798

andreas-kupries reviewed (approved):
Looks good now. Builds ok. Runs fine. No stacktraces for responding, but invalid endpoint
